### ISSUE TITLE
Clipboard manager no longer populating

### ISSUE THREAD

**Issue by curiousercreative** [wiiznokes/clipboard-manager#61]
#### Bug Description
Was working well for me for the past two weeks, but no longer works. Running latest source build, and latest cosmic packages (popdev) the applet is no longer capturing items copied to clipboard.

#### Expected Result
On copy, a new entry is added to the clipboard manager

#### Steps to Reproduce
1. Pull latest ([as of writing](https://github.com/wiiznokes/clipboard-manager/tree/cb21de7a0c01f61dc33112b24e1b8628781cf3bb)), build, and install
2. Reboot
3. Copy text from Firefox
4. Click the clipboard-manager applet to reveal the history
    ER: Text copied from FF appears at top of list
    AR: List is empty

#### Logs
```
$ cat /etc/os-release
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-o
```
```
$ ps aux | grep cosmic-ext-applet-clipboard-manager
curious+    9825  0.0  0.0 916820 23300 tty1     SNl+ 14:34   0:00 cosmic-ext-applet-clipboard-manager
```
```
$ cat /var/log/syslog | grep 'clipboard'
Jul 22 09:58:40 localhost cosmic-ext-applet-clipboard-manager[4752]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 09:58:40 localhost cosmic-ext-applet-clipboard-manager[4752]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 09:58:40 localhost cosmic-ext-applet-clipboard-manager[4752]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 09:58:40 localhost cosmic-ext-applet-clipboard-manager[4752]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 13:26:39 localhost cosmic-ext-applet-clipboard-manager[76705]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 13:26:42 localhost cosmic-ext-applet-clipboard-manager[76705]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 13:26:43 localhost cosmic-ext-applet-clipboard-manager[76705]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 13:26:43 localhost cosmic-ext-applet-clipboard-manager[77018]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 13:26:43 localhost cosmic-ext-applet-clipboard-manager[77018]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 13:26:43 localhost cosmic-ext-applet-clipboard-manager[77018]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 13:26:43 localhost cosmic-ext-applet-clipboard-manager[77018]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 14:19:36 localhost cosmic-ext-applet-clipboard-manager[86429]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 14:19:36 localhost cosmic-ext-applet-clipboard-manager[86429]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 14:19:36 localhost cosmic-ext-applet-clipboard-manager[86429]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 14:19:36 localhost cosmic-ext-applet-clipboard-manager[86429]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 14:20:19 localhost cosmic-ext-applet-clipboard-manager[86864]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 14:20:20 localhost cosmic-ext-applet-clipboard-manager[86864]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 14:20:20 localhost cosmic-ext-applet-clipboard-manager[86864]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 14:20:20 localhost cosmic-ext-applet-clipboard-manager[86864]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 14:21:51 localhost cosmic-ext-applet-clipboard-manager[88076]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 14:21:51 localhost cosmic-ext-applet-clipboard-manager[88076]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 14:21:51 localhost cosmic-ext-applet-clipboard-manager[88076]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 14:21:51 localhost cosmic-ext-applet-clipboard-manager[88076]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 15:47:05 localhost cosmic-ext-applet-clipboard-manager[101228]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 15:47:39 localhost cosmic-ext-applet-clipboard-manager[101228]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 15:47:39 localhost cosmic-ext-applet-clipboard-manager[101228]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 15:47:39 localhost cosmic-ext-applet-clipboard-manager[101228]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 15:47:40 localhost cosmic-ext-applet-clipboard-manager[101611]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 15:47:41 localhost cosmic-ext-applet-clipboard-manager[101611]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 15:47:41 localhost cosmic-ext-applet-clipboard-manager[101611]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 15:47:41 localhost cosmic-ext-applet-clipboard-manager[101611]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 23:57:32 localhost cosmic-ext-applet-clipboard-manager[3421]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 22 23:57:32 localhost cosmic-ext-applet-clipboard-manager[3421]: No `wp_viewporter`. Subsurfaces not supported.
Jul 22 23:57:32 localhost cosmic-ext-applet-clipboard-manager[3421]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 22 23:57:32 localhost cosmic-ext-applet-clipboard-manager[3421]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 10:03:32 localhost cosmic-ext-applet-clipboard-manager[4299]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 23 10:03:32 localhost cosmic-ext-applet-clipboard-manager[4299]: No `wp_viewporter`. Subsurfaces not supported.
Jul 23 10:03:32 localhost cosmic-ext-applet-clipboard-manager[4299]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 10:03:33 localhost cosmic-ext-applet-clipboard-manager[4299]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 12:16:49 localhost cosmic-ext-applet-clipboard-manager[39067]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 23 12:16:49 localhost cosmic-ext-applet-clipboard-manager[39067]: No `wp_viewporter`. Subsurfaces not supported.
Jul 23 12:16:49 localhost cosmic-ext-applet-clipboard-manager[39067]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 12:16:49 localhost cosmic-ext-applet-clipboard-manager[39067]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 12:37:47 localhost cosmic-ext-applet-clipboard-manager[3384]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 23 12:37:47 localhost cosmic-ext-applet-clipboard-manager[3384]: No `wp_viewporter`. Subsurfaces not supported.
Jul 23 12:37:47 localhost cosmic-ext-applet-clipboard-manager[3384]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 12:37:47 localhost cosmic-ext-applet-clipboard-manager[3384]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 12:38:11 localhost cosmic-ext-applet-clipboard-manager[5060]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 23 12:38:11 localhost cosmic-ext-applet-clipboard-manager[5060]: No `wp_viewporter`. Subsurfaces not supported.
Jul 23 12:38:11 localhost cosmic-ext-applet-clipboard-manager[5060]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 12:38:11 localhost cosmic-ext-applet-clipboard-manager[5060]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 14:34:41 localhost cosmic-ext-applet-clipboard-manager[9825]: Failed to initialize viewporter: the requested global was not found in the registry
Jul 23 14:34:41 localhost cosmic-ext-applet-clipboard-manager[9825]: inside error: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
Jul 23 14:34:41 localhost cosmic-ext-applet-clipboard-manager[9825]: No `wp_viewporter`. Subsurfaces not supported.
Jul 23 14:34:41 localhost cosmic-ext-applet-clipboard-manager[9825]: A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor
```
```
$ apt list cosmic-*
Listing... Done
cosmic-app-library/jammy,now 0.1.0~1721054203~22.04~3489c53 amd64 [installed]
cosmic-applets/jammy,now 0.1.0~1721314847~22.04~0c7083e amd64 [installed]
cosmic-bg/jammy,jammy,now 0.1.0~1720050061~22.04~343410f amd64 [installed]
cosmic-comp/jammy 0.1~1721743951~22.04~4f04313 amd64 [upgradable from: 0.1~1721316717~22.04~ec5d89e]
cosmic-design-demo/jammy 0.1.0~1704748391~22.04~d58cfad amd64
cosmic-edit/jammy,now 0.1.0~1721650308~22.04~33d6522 amd64 [installed]
cosmic-ext-orderly/jammy,jammy 0.1.0-0pop1~1719848956~22.04~d76bd39 amd64
cosmic-files/jammy 0.1.0~1721748812~22.04~1820433 amd64 [upgradable from: 0.1.0~1721224248~22.04~ec4d734]
cosmic-greeter-daemon/jammy,now 0.1.0~1721052517~22.04~59e2eaa amd64 [installed]
cosmic-greeter/jammy,now 0.1.0~1721052517~22.04~59e2eaa amd64 [installed]
cosmic-icons/jammy,now 0.1.0~1721234019~22.04~73be037 amd64 [installed]
cosmic-launcher/jammy,now 0.1.0~1721275594~22.04~c188029 amd64 [installed]
cosmic-notifications/jammy,now 0.1.0~1721052308~22.04~2caf2fb amd64 [installed]
cosmic-osd/jammy,now 0.1.0~1721052322~22.04~27fc5e5 amd64 [installed]
cosmic-panel/jammy,now 0.1.0~1721052062~22.04~05420b2 amd64 [installed]
cosmic-randr/jammy,now 0.1.0~1721287742~22.04~e214fe9 amd64 [installed]
cosmic-screenshot/jammy,now 0.1.0~1721052047~22.04~08ac28d amd64 [installed]
cosmic-session/jammy,now 0.1.0~1721052030~22.04~577a181 amd64 [installed]
cosmic-settings-daemon/jammy,now 0.1.0~1721272193~22.04~54700df amd64 [installed]
cosmic-settings/jammy 0.1.0~1721708515~22.04~4908e38 amd64 [upgradable from: 0.1.0~1721311726~22.04~e8106a9]
cosmic-store/jammy 0.1.0~1721663818~22.04~5d73d53 amd64 [upgradable from: 0.1.0~1721657927~22.04~34050c0]
cosmic-tasks/jammy,jammy 0.1.0-0pop1~1719848956~22.04~d76bd39 amd64
cosmic-term/jammy,now 0.1.0~1721650276~22.04~0652001 amd64 [installed]
cosmic-wallpapers/jammy,jammy,jammy,jammy,now 1.0.0~1719415313~22.04~0f2f16d all [installed,automatic]
cosmic-webapps/jammy,jammy,now 0.4.0-0pop1~1715817408~22.04~afb141a amd64 [installed]
cosmic-workspaces/jammy,now 0.1.0~1721334536~22.04~2de3669 amd64 [installed]
```

**Comment by wiiznokes** [comment]
> A required Wayland protocol (zwlr_data_control_manager_v1 version 1) is not supported by the compositor

This protocol is required to fetch clipboard entries. Did you install using `sudo just install` ? If yes, you should have the same output i have: for these commands:
```

fedasus@fedasus ~> echo $COSMIC_DATA_CONTROL_ENABLED
1
fedasus@fedasus ~> cat /etc/environment.d/cosmic-ext-applet-clipboard-manager.conf 
COSMIC_DATA_CONTROL_ENABLED=1
```

**Comment by wiiznokes** [comment]
@nbycoder can you report the output of `echo $COSMIC_DATA_CONTROL_ENABLED` ?

**Comment by wiiznokes** [comment]
btw, the file is now in `/usr/lib/environment.d/cosmic-ext-applet-clipboard-manager.conf`

**Comment by nbycoder** [comment]
@wiiznokes when i run  ```cat /usr/lib/environment.d/cosmic-ext-applet-clipboard-manager.conf``` it returns ```COSMIC_DATA_CONTROL_ENABLED=1``` if i run ```echo $COSMIC_DATA_CONTROL_ENABLED``` it doesn't return anything.

**Comment by wiiznokes** [comment]
That what i through, i seems ubuntu/pop-os doesn't respect this spec https://www.freedesktop.org/software/systemd/man/latest/environment.d.html

I'm open to suggestion to provide a cross distribution method for this

**Comment by wiiznokes** [comment]
Workaround: write `COSMIC_DATA_CONTROL_ENABLED=1` in `/etc/environment`. You will need to reboot so it take effect.

**Comment by nbycoder** [comment]
@wiiznokes Can confirm that worked for me! Thank you :)

**Comment by wiiznokes** [comment]
fixed by  #82
